Fix the failing tests

Since iamconsortium/common-definitions#336 is now merged, the tests that currently still fail can be updated:

```console
=========================== short test summary info ============================
FAILED tests/test_region_aggregation.py::test_mapping_from_external_repository - pydantic_core._pydantic_core.ValidationError: 1 validation error for RegionAggregationMapping
common_regions
  Name collision in common regions for ['Canada', 'Japan', 'China', 'Brazil', 'India'] in tests/data/region_processing/external_repo_test/common-definitions/mappings/GCAM_7.1.yaml [type=region_name_collision, input_value=[{'name': 'World', 'const...s': ['EU-12', 'EU-15']}], input_type=list]
FAILED tests/test_region_aggregation.py::test_mapping_from_external_repository_missing_regions_raises - AssertionError: Regex pattern did not match.
 Regex: 'validation errors for RegionProcessor'
 Input: "1 validation error for RegionAggregationMapping\ncommon_regions\n  Name collision in common regions for ['Canada', 'Japan', 'China', 'Brazil', 'India'] in tests/data/region_processing/external_repo_test_missing_region/common-definitions/mappings/GCAM_7.1.yaml [type=region_name_collision, input_value=[{'name': 'World', 'const...s': ['EU-12', 'EU-15']}], input_type=list]"
================= 2 failed, 230 passed, 61 warnings in 39.93s ==================
```

The `nomenclature.yaml` that describes the use of common-definitions pins a version prior to the fix. This pin should be updated to the latest version. This should fix the tests.

--- a/tests/data/region_processing/external_repo_test/nomenclature.yaml
+++ b/tests/data/region_processing/external_repo_test/nomenclature.yaml
@@ -4,7 +4,7 @@ dimensions:
 repositories:
   common-definitions:
     url: https://github.com/IAMconsortium/common-definitions.git/
-    hash: 091c0fe
+    hash: 082162b
 definitions:
   region:
     repository: common-definitions

--- a/tests/data/region_processing/external_repo_test_missing_region/nomenclature.yaml
+++ b/tests/data/region_processing/external_repo_test_missing_region/nomenclature.yaml
@@ -4,7 +4,7 @@ dimensions:
 repositories:
   common-definitions:
     url: https://github.com/IAMconsortium/common-definitions.git/
-    hash: 091c0fe
+    hash: 082162b
 definitions:
   region:
     repository:

